### ISSUE TITLE
1-to-1 multisig p2sh?

### ISSUE THREAD

**Issue by reardenlife** [bitcoinjs/bitcoinjs-lib#1371]
```js
function getScriptAddress (node) {
        const pubkeys = [ node.publicKey ];
  return bitcoin.payments.p2sh({
      redeem: bitcoin.payments.p2ms({ m: 1, pubkeys })
    }).address;
}
```

Is that the way to create the subject?

**Comment by dabura667** [comment]
yes. you should pass a network though, otherwise this function will only create bitcoin addresses.

**Comment by reardenlife** [comment]
> you should pass a network though, otherwise this function will only create bitcoin addresses.

Addresses on the mainnet you mean?
https://github.com/bitcoinjs/bitcoinjs-lib/blob/master/src/payments/p2sh.js#L49

Hm.. if this address is a hash of the script that should pay to the address derived from the public key provided, what my actions are in case if I am using something like electrum?

**Comment by junderw** [comment]
Sorry, could you rephrase your question?

I don't understand.

**Comment by reardenlife** [comment]
@junderw 
> "Sorry, could you rephrase your question?"

Well, yes.  Thanks for hearing me out.
I am trying to generate a bunch of addresses from a single seed mnemonic.
I already generated a bunch by feeding public keys to this function:

```js
function getAddress (node, network) {
  return bitcoin.payments.p2pkh({ pubkey: node.publicKey, network }).address
}
```

On the output I've got the addresses starting with 1.
It is more or less clear how to work with such addresses in some bitcoin client such as electrum: http://docs.electrum.org/en/latest/faq.html . I can "sweep" or import the private keys I generated the addresses from.  After that the electrum will monitor the blockchain and notify me if someone send funds to the addresses I imported.

Now I am trying to do the same, but for p2sh addresses, those thar start with 3.
I am generating 1-to-1 multisig p2sh addresses and I just want to do the same thing - to receive funds.
How exactly it should be done? Should I provide the script to the electrum or ... ?
